### PR TITLE
feat: add the cutoff distances for contact features

### PR DESCRIPTION
**Pull request by cbaakman**
Use cutoff distances for calculating vanderwaals and electrostatic energies between atoms. Just like in the old deeprank.

### CODE REVIEW

File: deeprankcore/features/contact.py [hunk 1]
```diff
@@ -13,21 +13,28 @@
...
-def _get_coulomb_potentials(atoms: List[Atom], distances: np.ndarray) -> np.ndarray:
+def _get_coulomb_potentials(atoms: List[Atom], distances: np.ndarray, cutoff_distance: float) -> np.ndarray:
     """ 
         Calculate Coulomb potentials between between all Atoms in atom.
         Warning: there's no distance cutoff here. The radius of influence is assumed to infinite.
             However, the potential tends to 0 at large distance.
     """
 
+    clamped_distances = distances
+    clamped_distances[distances == 0.0] = 3.0
```
Comment (gcroci2) [inline deeprankcore/features/contact.py:24]: From where does this come from? (value of 3 for 0 distances)

File: deeprankcore/features/contact.py [hunk 2]
```diff
@@ -13,21 +13,28 @@
...
+
     EPSILON0 = 1.0
     COULOMB_CONSTANT = 332.0636
     charges = [atomic_forcefield.get_charge(atom) for atom in atoms]
-    coulomb_potentials = np.expand_dims(charges, axis=1) * np.expand_dims(charges, axis=0) * COULOMB_CONSTANT / (EPSILON0 * distances)
-    return coulomb_potentials
+    coulomb_potentials = np.expand_dims(charges, axis=1) * np.expand_dims(charges, axis=0) * COULOMB_CONSTANT / (EPSILON0 * clamped_distances)
+
+    cutoff_factors = (1.0 - (clamped_distances / cutoff_distance) ** 2) ** 2
```
Comment (gcroci2) [inline deeprankcore/features/contact.py:31]: Where did you take this new formula?

**Review by gcroci2 — COMMENTED** [review]
I think it would be great if @LilySnow reviews this, just to be sure that the potentials are computed as she meant.

File: deeprankcore/features/contact.py (hunk 1 above)
Comment (cbaakman) [reply deeprankcore/features/contact.py:24]: This came from:
https://github.com/DeepRank/deeprank/blob/3594e05dca15e688c10ef66e0fbc7cd71fd40027/deeprank/features/AtomicFeature.py#L617

File: deeprankcore/features/contact.py (hunk 2 above)
Comment (cbaakman) [reply deeprankcore/features/contact.py:31]: It was here:
https://github.com/DeepRank/deeprank/blob/3594e05dca15e688c10ef66e0fbc7cd71fd40027/deeprank/features/AtomicFeature.py#L620

File: deeprankcore/features/contact.py [hunk 3]
```diff
@@ -40,15 +47,26 @@ def _get_lennard_jones_potentials(atoms: List[Atom], distances: np.ndarray) -> n
...
+
     # calculate inter potentials
     sigmas = [atomic_forcefield.get_vanderwaals_parameters(atom).inter_sigma for atom in atoms]
     epsilons = [atomic_forcefield.get_vanderwaals_parameters(atom).inter_epsilon for atom in atoms]
     mean_sigmas = 0.5 * np.add.outer(sigmas,sigmas)
     geomean_eps = np.sqrt(np.multiply.outer(epsilons,epsilons))     # sqrt(eps1*eps2)
     inter_potentials = 4.0 * geomean_eps * ((mean_sigmas / distances) ** 12 - (mean_sigmas / distances) ** 6)
 
-    lennard_jones_potentials = {'intra': intra_potentials, 'inter': inter_potentials}
+    # calculate prefactors
```
[the remark this answers is not in the document]
Comment (cbaakman) [reply deeprankcore/features/contact.py:58]: From here:
https://github.com/DeepRank/deeprank/blob/3594e05dca15e688c10ef66e0fbc7cd71fd40027/deeprank/features/AtomicFeature.py#L899

**Comment by cbaakman** [comment]
> I think it would be great if @LilySnow reviews this, just to be sure that the potentials are computed as she meant.

Quite right! Sorry I put this all on your desk.

**Review by gcroci2 — APPROVED** [review]

**Comment by gcroci2** [comment]
Please fix the checks before merging :) @cbaakman

**Comment by DaniBodor** [comment]
@cbaakman does this PR relate to the discussion in #357 or is it a completely separate issue?

**Comment by cbaakman** [comment]
> @cbaakman does this PR relate to the discussion in #357 or is it a completely separate issue?

Indeed! Before merging this pull request, we must implement a solution for this discussion.

**Comment by cbaakman** [comment]
It's no longer relevant, because it's been replaced.

**Pull request closed without merging**